modules/entity_to_text_tika/tests/src/Kernel/PlaintextStorageTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Drupal\Tests\entity_to_text_tika\Kernel;
+
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\entity_to_text_tika\Storage\PlaintextStorage;
+use Drupal\file\Entity\File;
+use Drupal\KernelTests\Core\File\FileTestBase;
+
+/**
+ * Tests the Plaintext File Storage.
+ *
+ * @coversDefaultClass \Drupal\entity_to_text_tika\Storage\PlaintextStorage
+ *
+ * @group entity_to_text
+ * @group entity_to_text_tika
+ *
+ * @internal
+ */
+final class PlaintextStorageTest extends FileTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = ['file', 'entity_to_text_tika', 'user'];
+
+  /**
+   * The file system service.
+   *
+   * @var \Drupal\Core\File\FileSystemInterface
+   */
+  protected $fileSystem;
+
+  /**
+   * The plain-text storage processor.
+   *
+   * @var \Drupal\entity_to_text_tika\Storage\PlaintextStorage
+   */
+  protected PlaintextStorage $plaintextStorage;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installEntitySchema('file');
+    $this->installEntitySchema('user');
+    $this->installSchema('file', ['file_usage']);
+
+    $this->fileSystem = $this->container->get('file_system');
+    $this->plaintextStorage = $this->container->get('entity_to_text_tika.storage.plain_text');
+
+    $destination = PlaintextStorage::DESTINATION;
+    $this->fileSystem->prepareDirectory($destination, FileSystemInterface::CREATE_DIRECTORY | FileSystemInterface::MODIFY_PERMISSIONS);
+  }
+
+  /**
+   * @covers ::loadTextFromFile
+   */
+  public function testLoadTextFromFile(): void {
+    // Create an OCR file for testing.
+    $file_uri = $this->createUri('390-foo.txt.en.ocr.txt', 'Ipsum excepteur id cupidatat commodo', 'private');
+    $this->fileSystem->move($file_uri, 'private://entity-to-text/ocr/390-foo.txt.en.ocr.txt', FileSystemInterface::EXISTS_REPLACE);
+
+    // Create a file that correspond to the previous OCR file.
+    $file = File::create([
+      'uri' => 'public://foo.txt',
+      'name' => 'foo',
+    ]);
+    $file->set('fid', 390);
```

Comment:
Could you add two more tests:

1. when the File is private (`private://foo.txt`) : does it make a collision with `public://foo.txt` ?
2. when the File is in a subdirectory : `public://documents/2024/foo.txt`